preparedstatement/src/test/java/software/amazon/athena/preparedstatement/DeleteHandlerTest.java

```diff
@@ -0,0 +1,89 @@
+package software.amazon.athena.preparedstatement;
+
+import java.time.Duration;
+import software.amazon.awssdk.services.athena.AthenaClient;
+import software.amazon.awssdk.services.athena.model.DeletePreparedStatementRequest;
+import software.amazon.awssdk.services.athena.model.DeletePreparedStatementResponse;
+import software.amazon.awssdk.services.athena.model.GetPreparedStatementRequest;
+import software.amazon.awssdk.services.athena.model.GetPreparedStatementResponse;
+import software.amazon.awssdk.services.athena.model.PreparedStatement;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class DeleteHandlerTest extends AbstractTestBase {
+
+    @Mock
+    private AmazonWebServicesClientProxy proxy;
+
+    @Mock
+    private ProxyClient<AthenaClient> proxyClient;
+
+    @Mock
+    AthenaClient athenaClient;
+
+    @BeforeEach
+    public void setup() {
+        proxy = new AmazonWebServicesClientProxy(logger, MOCK_CREDENTIALS, () -> Duration.ofSeconds(600).toMillis());
+        athenaClient = mock(AthenaClient.class);
+        proxyClient = MOCK_PROXY(proxy, athenaClient);
+    }
+
+    @AfterEach
+    public void tear_down() {
+        verify(athenaClient, atLeastOnce()).serviceName();
+        verifyNoMoreInteractions(athenaClient);
+    }
+
+    @Test
+    public void handleRequest_SimpleSuccess() {
```

Comment:
Is there any other cases we would want to test here? Maybe when the resource has already been deleted

Reply:
sure, I can add a test case for that.